src/actions/actions.ts

```diff
@@ -1960,6 +1960,19 @@ class ActionDeleteChar extends BaseCommand {
 }
 
 @RegisterAction
+class ActionDeleteCharWithDeleteKey extends ActionDeleteChar {
+  keys = ["<delete>"];
+
+  public async execCount(position: Position, vimState: VimState): Promise<VimState> {
+    if (vimState.recordedState.count !== 0) {
```

Comment:
The fact that Ndelete does nothing is so bizarre that I totally misread this. This is fine.